src/azul/queues.py

```diff
@@ -79,7 +79,7 @@ def dump_all(self):
 
     def _dump(self, queue, path):
         logger.info('Writing messages from queue %r to file %r', queue.url, path)
-        messages = self._get_messages(queue)
+        messages = self.read_messages(queue)
```

Comment:
I think this implementation first reads the messages from the queue, then deletes them, and then writes the messages to disk. If the writing fails, the messages are lost forever. The original implementation deleted the messages after writing them.